Drop-In/src/androidTest/java/com/braintreepayments/api/AddCardFragmentUITest.kt

```diff
@@ -0,0 +1,137 @@
+package com.braintreepayments.api
+
+import androidx.fragment.app.testing.FragmentScenario
+import androidx.lifecycle.Lifecycle
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.action.ViewActions.typeText
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.*
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.braintreepayments.api.CardNumber.VISA
+import com.braintreepayments.api.dropin.R
+import com.braintreepayments.cardform.utils.CardType
+import junit.framework.TestCase.assertNull
+import org.hamcrest.CoreMatchers.not
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.util.concurrent.CountDownLatch
+
+@RunWith(AndroidJUnit4::class)
+class AddCardFragmentUITest {
+
+    private lateinit var countDownLatch: CountDownLatch
+
+    @Before
+    fun beforeEach() {
+        countDownLatch = CountDownLatch(1)
+    }
+
+    @Test
+    fun whenStateIsRESUMED_buttonTextIsAddCard() {
+        val scenario = FragmentScenario.launchInContainer(AddCardFragment::class.java, null, R.style.bt_drop_in_activity_theme)
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        scenario.onFragment { fragment ->
+            fragment.dropInViewModel.setSupportedCardTypes(listOf(CardType.VISA))
+        }
+
+        onView(isRoot()).perform(waitFor(500))
+        onView(withId(R.id.bt_button)).check(matches(withText(R.string.bt_next)))
```

Comment:
Should the tests be using the R id values or hard coded string values 🤔 . The strings are definitely easier to read. Also wouldn't using the raw strings catch issues if for some the R id values changed - our tests would break to catch it?

Reply:
Yeah we're doing it like this because most of the Espresso test runner tutorials use this method. I agree, it's probably better overall to go with the actual text. We really wanted to get everything modernized and broken up into fragments, then we'll probably refine the testing strategy a bit more.

Reply:
👍

Reply:
Should we leave a TODO?

Reply:
Done!